pkg/skaffold/build/build.go

```diff
@@ -30,14 +30,24 @@ type Artifact struct {
 	Tag       string `json:"tag"`
 }
 
+// Result contains the result of a build of a single target artifact.
+// It contains the target artifact (parsed from the skaffold config),
+// and either an Artifact representing the result of a successful build,
+// or the error encountered while executing the build.
+type Result struct {
+	Target latest.Artifact
+	Result Artifact
+	Error  error
+}
+
 // Builder is an interface to the Build API of Skaffold.
 // It must build and make the resulting image accesible to the cluster.
 // This could include pushing to a authorized repository or loading the nodes with the image.
 // If artifacts is supplied, the builder should only rebuild those artifacts.
 type Builder interface {
 	Labels() map[string]string
 
-	Build(ctx context.Context, out io.Writer, tags tag.ImageTags, artifacts []*latest.Artifact) ([]Artifact, error)
+	Build(ctx context.Context, out io.Writer, tags tag.ImageTags, artifacts []*latest.Artifact) ([]chan Result, error)
```

Comment:
I wonder - why don't we just return a single channel of results?

Reply:
theoretically we could just share one big return channel with capacity len(artifacts)